app/controllers/application_controller.rb

```diff
@@ -5,10 +5,31 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
+  protected
   # Use callbacks to share common setup or constraints between actions.
   def set_vault
-    @vault = params[:vault_id] ? Vault.find_by_id(params[:vault_id]) : Vault.find_by_id(params[:id])
-    redirect_to root_url if @vault.nil? || @vault.user_username != params[:username]
+    @vault = params[:vault_id] ? Vault.find_vault(params[:vault_id]) : Vault.find_vault(params[:id])
+  end
+
+  def authenticate_username
+    user = User.find_by_username(params[:username])
+    redirect_to root_url if user.nil? || user.vaults.find_vault(params[:vault_id]).nil?
+  end
+
+  def check_edit_permissions
+    if @vault.user_id != current_user.id then
```

Comment:
[Style guide](https://github.com/bbatsov/ruby-style-guide#no-then) prefers omitting the `then`.